_pages/mobile-interview-project.md

```diff
@@ -89,63 +144,108 @@ Feel free to either:
 
-Please **do not** use code from previous/current employers, for obvious reasons :-)
+Sort of: phones and tablets are pretty fast these days, so there’s no need to prematurely optimize anything or do microbenchmarks. Please do try to make sure that there aren’t any problems when using the app (like dropped frames, freezing).
 
-**Q) What programming languages can I use?**
-Please utilize Java or Kotlin for Android, or Objective-C or Swift for iOS. For iOS, if you need to drop down to CoreFoundation, Accelerate, etc, C is fine too (but you shouldn’t need to).
+That being said, do consider network usage: we will want to lazy load images in order to avoid wasting the user’s data plan, and not call the employee list service more than necessary.
 
-Please **do not** utilize cross-platform SDKs such as React Native, Flutter, Phonegap, Cordova, etc.
+ * Q) **For iOS, can I use xibs and storyboards?**
 
-**Q) Can I use 3rd party frameworks?**
-Sure – feel free to use any “no brainer” frameworks you’d usually pull into an app such as RxJava, RxSwift, etc. However, since this is a relatively simple application, you shouldn’t need many external dependencies. In particular, we’d like to see no 3rd party image caching framework used for loading and storing employee’s profile images (since this is an interesting part of the question).
+Feel free to do so. However, note that we do not (yet) write code in Swift UI or Compose at Square. If you use the Compose beta, please let us know how to build and run your app.
 
-...And of course, 1st-party frameworks that come with iOS or Android are fine.
+ * Q) **Can I use SwiftUI (iOS) or Compose (Android)?**
 
-**Q) What build systems & build tooling should I use?**
-Please utilize platform-standard build systems and tooling. If you decide to use tooling such as CocoaPods or Bazel to assist in project configuration and set up, include a README explaining how to build and run the app.
+Feel free to do so. However, note that we do not (yet) write code in Swift UI or Compose at Square. If you use the Compose beta, please let us know how to build and run your app.
 
-**Q) What version of iOS or Android SDKs and tooling should I use?**
-Please utilize modern tooling and development practices. For this case, consider anything released in the last 1 year as “modern”.
+ * Q) **Can I use Data Binding (Android?)**
 
-**Q) What are you looking for? What is the right answer?**
-There isn’t one right answer! We’re looking to see how you approach the problem. The generic things we’re evaluating are architecture and structure of your code, how maintainable and testable it is, isolation of concerns, etc.
+Thanks for keeping up to date with the latest Android technologies! In that spirit, since Data Binding has been softly discouraged by Android in favor of View Binding, and since we don’t use Data Binding at Square and wouldn’t be as effective in reviewing its usage, please avoid using Data Binding in your submissions.
```

Comment:
I think we should consider removing this.  Unless I'm mistaken, Google has never officially discouraged it.  I think if you asked George Mount prior to Compose, he would have encouraged it even :-)  Even after ViewBinding came out, Google still added support for Lifecycle and LiveData to DataBinding.  And many companies still use it.    

Without a doubt, DataBinding is controversial, similar to Dagger vs Koin vs KodeIn, etc.  But I don't see why we should discourage or call it out.  If the candidate wishes to use it and can explain why they chose to use it, it's strengths and its weaknesses, then it's a conversation worth having (and IMO not a dent against the candidate).

This also seems like a "one off", in the sense that there are a lot of tech choices we could mention here.  For example, I think we could say the same about RxJava/RxKotlin.

```
* Q) **Can I use RxJava with Kotlin on (Android?)**

Thanks for keeping up to date with the latest Android technologies! In that spirit, since RxJava has been softly discouraged by Google and JetBrains in favor of Coroutines/Flow, and since many teams at Square don't use RxJava we wouldn't be as effective in reviewing its usage, please avoid using RxJava/RxKotlin in your submissions.
```

Reply:
I would disagree with your comment with the simple problem that since we intentionally don't use it at Square we won't be able to review it.

We can remove the portions about being discouraged by Google though.

Reply:
That would be a good compromise.  But i still think this feels out of place. A few more examples to consider 

We also intentionally don’t use MVVM, MvRx, MVP or any other UI framework intentionally because we feel Workflow is better but we don’t dictate that and it doesn’t prevent us from understanding the candidates submission. 

Similarly, we don’t use Room because we think SQLDelight is more appropriate.  But we don’t prevent candidates from using Room or Realm.  

Regarding interviewer familiarity, some candidates choose Compose and many use Hilt, both of which we are not familiar with and still we are able to get a good evaluation of the candidate.  

This feels like a very slippery slope to me.  It seems like we should use a tech list in this case of approved technologies that we approve for use. 

If that doesn’t sway other opinions, then let’s leave it but can we consider dropping the Google soft discouraging (as you mentioned) and also drop the part about it being new technology?

Reply:
I'm struggling with this one as well. A candidate may have a technically defensible position for using databinding. We also contradict our statement "Of course, use 1st party libraries!". With squareism in mind, would we subtract for using fragments? I don't think using databinding will create an impossible review scenario for square reviewers, and it does feel more fair to me at least that we allow candidates to demonstrate their current architectures and views.

Reply:
I'm struggling with this one as well. A candidate may have a technically defensible position for using databinding. We also contradict our statement "Of course, use 1st party libraries!". With squareism in mind, would we subtract for using fragments? I don't think using databinding will create an impossible review scenario for square reviewers, and it does feel more fair to me at least that we allow candidates to demonstrate their current architectures and views.

Reply:
I think if I saw a project that used DataBinding correctly and they're able to talk about it, it wouldn't be a red flag for me, just something to note. After reading this thread, I'm leaning towards not calling out data binding at all.